Detectors/Base/include/DetectorsBase/Vector.h

```
@@ -0,0 +1,384 @@
+  
+ protected:
+
+  // offset of data wrt buffer start: account for Header + padding after Header block to respect T alignment
+  constexpr sizeType dataOffset() const {return sizeof(Header) + sizeof(Header)%alignof(T);};
+  // for some reason using static constexpr leads to problem in linking the testVector
+  //  static constexpr sizeType dataOffset() = sizeof(Header) + sizeof(Header)%alignof(T);
+
+  
+  std::unique_ptr<char[]> mPtr;                //! pointer on continuos block containing full object data
+  std::vector<T> mVecForTree;                  //! vector for writing objects into the tree
+  
+  ClassDef(Vector,1)
+};
+
+
+
+//-------------------------------------------------------------------
+template <class T, class H>
+  Vector<T,H>::Vector(sizeType iniSize, int expPol) : mPtr(nullptr), mVecForTree()
+  {
+    /**
+     * Creates container for objects of 
+     * T POD class
+     * with single description field of H POD class
+     * size: initial capacity of the container
+     * expPol: expansion policy. new_size = expPol>0 ? oldSize+expPol : 2*max(1,oldSize+|expPol|)
+     */
+    static_assert(std::is_trivially_copyable<T>::value,"Object class is not trivially-copiable");
+    static_assert(std::is_trivially_copyable<H>::value,"Header class is not trivially-copiable");
+    reserve(iniSize);
+    if (!expPol) expPol = -1;
+    getHeader()->expandPolicy = expPol;
+  }
+
+//-------------------------------------------------------------------
+template <class T, class H>
+  void Vector<T,H>::recreate(const char* ptr, sizeType nbytes)
+{
+  /**
+   * recreates container buffer from provided raw pointer 
+   * ptr: pointer extracted from the container of simular type and owned by it, using e.g. getPtr() method.
+   * nbytes: expected size of the buffer in bytes, used for error check
+   */
+
+  if (ptr==nullptr) {
+    std::ostringstream strErr;
+    strErr << "invalid arguments: ptr is null" << std::endl;
+    throw strErr.str();
```

Why not simply `throw std::runtime_error("invalid arguments: ptr is null")`?